guidance/models/_azure_guidance.py

```diff
@@ -0,0 +1,167 @@
+import requests
+import os
+import base64
+import json
+import urllib.parse
+from ._model import Engine, Model, EngineCallResponse
+from .._chat import Phi3ChatTemplate
+from ._byte_tokenizer import ByteTokenizer
+
+
+class AzureGuidanceEngine(Engine):
+    """This connects to a remote guidance server on Azure and runs all computation using the remote engine."""
+
+    def __init__(self, server_url, max_streaming_tokens=1000, chat_template=None):
```

Comment:
How is auth handled?

Reply:
currently in the URL, but this is something that should change before the release.